frontend/components/common/InstantSearchProvider.tsx

```diff
@@ -217,6 +220,33 @@ export function InstantSearchProvider({
                   ? filter
                   : {};
 
+            Object.entries(filterObject).forEach(([attribute, value]) => {
```

Comment:
Would it be simpler to define the type of the whole `filterObject` using `zod` here instead of writing code that does the same?

Reply:
That would be nice. I had considered that, but I struggled to imagine what the implementation would look like.

**Option A:** Hard code a zod schema with all the facet names
```js
zod.object({
   "facet_tags.Item Type": z.string(),
   "facet_tags.Price Range": z.string().array(),
   ...,
});
```
Con: This schema could get out of sync with `useFacets()`.

**Option B:** Dynamically generate a zod schema with the result of `useFacets()`

Context: The `filterObject` will include any other garbage that happens to be under the `filter` namespace in the query params. For example, `filter[asdf]=qwer` in the params would generate `{ asdf: "qwer" }` in the filter object.

Algolia ignores any filter params whose key (aka attribute) don't match a real facet.

However, in order to dynamically generate the schema we'd need to be able to determine:
- if each attribute in `filterObject` is a valid facet
- if each valid attribute in `filterObject` is a Menu or a RefinementList

All that seems like it would require just as much code as validating the `filterObject` ourselves, plus we get the benefit of transforming the invalid menu params so that they are correctly applied when the page loads.

I'd be happy to implement this in a followup PR if we can come up with a solution tho